packages/cozy-procedures/src/components/overview/helpers.js

```diff
@@ -0,0 +1,86 @@
+import formatDate from 'date-fns/format'
+import subMonths from 'date-fns/sub_months'
+import startOfMonth from 'date-fns/start_of_month'
+import pick from 'lodash/pick'
+
+/**
+ * createJsonFile - upload a json file containing the given data
+ *
+ * @param {string} filename - The json file name
+ * @param {string} destinationId - The id of the destination directory
+ * @param {string} content - The file content
+ * @return {object} Information about the jsonFile (id and name)
+ */
+export const createJsonFile = async (
+  client,
+  filename,
+  destinationId,
+  content
+) => {
+  const file = new File([content], filename)
+  const resp = await client
+    .collection('io.cozy.files')
+    .createFile(file, { dirId: destinationId })
+
+  return resp.data
+}
+
+export const fetchCheckingsAccounts = async client => {
+  const query = client.find('io.cozy.bank.accounts').where({
+    type: {
+      $in: ['Checkings', 'Bank', 'Cash', 'Deposit']
+    }
+  })
+
+  // TODO extract this loop to cozy-client
+  const accounts = []
+  let resp = { next: true }
+
+  while (resp && resp.next) {
+    resp = await client.query(query.offset(accounts.length))
+    accounts.push(...resp.data)
+  }
+
+  return accounts
+}
+
+export const fetchTransactionsHistory = async (client, accountsIds) => {
+  const today = startOfMonth(new Date())
+  const threeMonthsBefore = startOfMonth(subMonths(today, 3))
+
+  const query = client.find('io.cozy.bank.operations').where({
+    account: { $in: accountsIds },
+    date: {
+      $lt: formatDate(today, 'YYYY-MM-DD'),
+      $gt: formatDate(threeMonthsBefore, 'YYYY-MM-DD')
+    }
+  })
+
+  // TODO extract this loop to cozy-client
+  const transactions = []
+  let resp = { next: true }
+
+  while (resp && resp.next) {
+    resp = await client.query(query.offset(transactions.length))
+    transactions.push(...resp.data)
+  }
+
+  return transactions
+}
+
+export const fetchTransactionsHistoryContent = async client => {
+  const accounts = await fetchCheckingsAccounts(client)
+  const accountsIds = accounts.map(account => account._id)
+  const transactions = await fetchTransactionsHistory(client, accountsIds)
+
+  const rawContent = accounts.map(account => {
+    return {
+      ...pick(account, ['balance', 'iban']),
+      transactions: transactions
+        .filter(transaction => transaction.account === account._id)
```

Comment:
Nit: if fetchTransactionsHistory did a groupBy account id when returning results (or if the groupBy was done L75), we would not have to filter n times on the same list here.